src/ProductView/ProductView.js

```diff
@@ -121,12 +161,30 @@ class ProductView extends Component {
             this.setState({filter: updatedFilters});
         }
     }
-    
+
     setActiveEventType(eventType) {
         this.fetchEventsFor(eventType);
         this.setState({activeEventType: eventType});
     }
-    
+
+    setProductConfiguration(configurationId) {
+        this.setState({
+            loading: true
+        });
+        if(configurationId) {
+            this.setState({
```

Comment:
What should be overwritten? First, 'loading' is set, later only 'showAllConfigurations' in state is touched and I don't rely on the state of 'loading'. Therefore nothing should get overwritten.
React Docu says, that it may batch the 'setState'-calls, but I don't see how this should affect this function in a bad way :)

> There is no guarantee of synchronous operation of calls to setState and calls may be batched for performance gains.